text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+    This command triggers the creation of an enclave and registers it at the FPC Registry.  
+    The registration includes the attestation of the Chaincode Enclave.
+  -->
+  For more information on additional management commands, see the [FPC Management API document](https://github.com/hyperledger-labs/fabric-private-chaincode/blob/master/docs/design/fabric-v2%2B/fpc-management.md).
+
+<!-- commented out as not really user visible
+The FPC Registry stores all attestation reports, as signed by the TEE vendor. 
+The attestation reports include information on what chaincode is running; what specific TEE is in use, including version number and hash of the chaincode (MRENCLAVE), and including the public keys of the enclave. 
+Upon creation each chaincode enclave generates public/private key pairs for signing and for encryption. The public signing key also denotes the chaincode enclave identity.
+
+In order to support the endorsement model of Fabric, chaincode enclaves executing the same FPC Chaincode need access to a set of shared secrets. In particular, these shared or common keys are used to encrypt and decrypt the transaction arguments and state.
+That is, there is also a public/private key pair and a symmetric state encryption key per FPC Chaincode that are shared among all chaincode enclaves that run the same FPC Chaincode using the key distribution protocol.
+-->
+
+A detailed description (internal view) of the FPC deployment process is provided in the FPC 1.0 Architecture section below and can be found in the [Full Detail Diagrams](#full-detail-diagrams) section.
+
+
+# Threat & Trust Model
+<!-- note: put threat-model here as threat-model is crucial to understand when looking at architecture and the application classes covered, all of this only covered here.  
+So putting it before User Experience would put it a bit in wrong context ...
+Arguably, it might even be better to put it _after_ the architecture as there are essentially already some forward references to architecture features involved.
+-->
+
+- Organizations do not have to trust each other (i.e., users, admins and software such as peers of another organization or arbitrary collusions thereof) as far as confidentiality is concerned.
+An organization can modify any software (including any hypervisor, operating system, or Fabric itself).
+Yet, such an organization would not be able to extract private state from the chaincode,
+or learn anything about the requests or responses of the victim organization (other than what the chaincode logic allows them to learn about it).
+
+- Organizations do have to trust a quorum of other organizations as defined by the lifecycle and chaincode endorsement policies as far as integrity of chaincode metadata is concerned. E.g., they have to rely on quorums of admins to only create and modify chaincodes as appropriate and they have to rely on quorum of peers to properly execute enclave registry and validation chaincode transactions.
```

Comment:
Can you clarify whether "chaincode endorsement policies" applies to lifecycle versus the actual FPC chaincode?
There would still be a endorsement policy for the FPC chaincode right? But it could safely indicate that only a single org would have to endorse?

```suggestion
- Organizations do have to trust a quorum of other organizations as defined by the lifecycle and chaincode endorsement policies as far as integrity of chaincode metadata is concerned. E.g., they have to rely on quorums of admins to only create and modify chaincodes as appropriate and they have to rely on quorum of peers to properly execute enclave registry and validate chaincode transactions.
```

Reply:
The endorsement policy as defined as part of the chaincode definition applies to enclave endorsement validation. (Recall that an FPC invocation is composed by a `peer chaincode query` to execute the chaincode and get the enclave endorsement, and a `peer chaincode invocation` to execute the enclave endorsement validation.) That is, with FPC we use the normal chaincode EP to describe how many Orgs are required to validate an enclave endorsement. 
This is also mentioned in the "Deploy a FPC Chaincode Package" Section.

For FPC 1.0 we indeed propose a single Org/Enclave endorsement and added support for multiple enclaves and richer FPC chaincode endorsement policies to our roadmap.

Reply:
ok, could you clarify this in the RFC.

Reply:
@denyeart we clarified this in the RFC.